### ISSUE TITLE
[Bug][Chart] Inconsistent zoom on selection behavior

### ISSUE THREAD

**Issue by vveesseelliinnaa** [telerik/kendo-react#2434]
## I'm submitting a...

* Bug report 

## Current behavior
Currently, the zoom on select (pressing Shift) gets reset if you either start the selection from the right to the left, or if you end the selection outside the chart area.

Video: https://somup.com/cZjh2HGuNw

## Expected behavior
The zooming behavior should be the same in all scenarios.

## Minimal reproduction of the problem with instructions

1. Got to https://www.telerik.com/kendo-react-ui/components/charts/chart/pan-zoom/
2. Press Shift and start a zooming selection from the bottom right part of the chart moving to the left top one.

## Additional details:
Reported in Ticket ID: 1660404
Private Issue - https://github.com/telerik/kendo-charts/issues/488

**Comment by pavlovaconsultingbv** [comment]
Hi Vesseeeliinnaaa ;) 

It's interesting how you've brought up two additional points I overlooked. 
I just tested the two bugs reported by you and indeed that behaviour is not optimal, but.. we could live with them. 

3 other undesired behaviours/bugs exist. They are all similar to the bug you noticed: that the zoom is executing  "if you end the selection outside the chart area".
Unfortunatelly, the 3 bugs below make the Zoom feature 100% unusable for us and the zoom is a key feature for our analytic product. 
Here they come: 

BUG 1. The zoom is executing once you move your mouse up
Currently, when the user presses shift and start is dragging the zoom selection in the south-east direction AND while doing this moves the pointer up even a little bit the zoom executes. 
Expected behavior:  moving the pointer up should not execute the zoom, but just decrease the selection area north direction. 
In general: **As long as the user holds the shift + mouse clicked the zoom should not execute**

BUG 2. The zoom is executing once you touch the X axis
Currently, when the user presses shift and drags the zoom selection in the south-east direction AND reaches X axis after a moment the zoom executes
Expected behavior:  
moving the pointer below the X axis should not execute the zoom
In general: **As long as the user holds the shift + mouse clicked the zoom should not execute**

BUG 3. The zoom is executing one it touches a data point (or for no reason)

Currently, when the user presses shift and drags the zoom selection in the south-east direction after some time the zoom self-executes. (I cannot really pinpoint when this happens exactly.. maybe when it touches some points at the bars OR after X seconds ?? )
Expected behavior:  **As long as the user holds the shift + mouse clicked the zoom should not execute**

**To sum up** 
The bug found by you and all the 3 critical bugs above can be solved by the same correction:
**As long as the user holds the shift + mouse clicked the zoom should not execute**

I am the user and I want to play, move the cursor up left, right, below the X axis, maybe take 1 minute to decide. I may be doing all kind of strange things when I keep that shift + mouse clicked. 
... but I decide when I finish the selection and I will then release the mouse click. No need to help me with this ;) 

I hope this helps. 
Best, 
Lukasz
